Packs/Illumio/Integrations/IllumioCore/IllumioCore_test.py

```diff
@@ -38,3 +55,224 @@ def test_test_module(requests_mock, mock_client):
+      'end_time': "2022-07-18T12:58:33.529Z", 'policy_decisions': ['potentially_blocked']},
+     "{} is an invalid value for {}. Possible values are: {}.".format('dummy', 'protocol', VALID_PROTOCOLS))
+])
+def test_illumio_traffic_analysis_command_for_invalid_arguments(args, err_msg, mock_client):
+    """
+    Test case scenario for execution of illumio-traffic-analysis-command when invalid argument provided.
+
+    Given:
+        - command arguments for illumio_traffic_analysis_command
+    When:
+        - Calling `illumio_traffic_analysis_command` function
+    Then:
+        - Returns a valid error message
+    """
+    from IllumioCore import illumio_traffic_analysis_command
+    with pytest.raises((ValueError, IllumioException, InvalidPortException, InvalidSingleSelectException,
+                        InvalidMultiSelectException)) as err:
+        illumio_traffic_analysis_command(mock_client, args)
+    assert str(err.value) == err_msg
+
+
+def test_illumio_traffic_analysis_success(requests_mock, mock_client):
+    """
+    Test case scenario for successful execution of illumio_traffic_analysis_command function.
+
+    Given:
+        - command arguments for illumio_traffic_analysis
+    When:
+        - Calling `illumio_traffic_analysis_Command` function
+    Then:
+        - Returns a valid output
+    """
+    from IllumioCore import illumio_traffic_analysis_command
+
+    args = {'port': 8443, 'protocol': "tcp", 'policy_decisions': 'potentially_blocked',
+            'start_time': "2022-07-17T12:58:33.528Z", 'end_time': "2022-07-18T12:58:33.529Z"}
+
+    json_data = util_load_json(os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                                            "test_data/traffic_analysis_success_response.json"))
+
+    with open(os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                           "./test_data/traffic_analysis_success_hr.md")) as file:
+        hr_output = file.read()
+
+    json_data_get = util_load_json(os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                                                "test_data/traffic_analysis_get_response.json"))
+
+    json_download_data = util_load_json(os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                                                     "test_data/traffic_analysis_download_data.json"))
+
+    requests_mock.post("https://127.0.0.1:8443/api/v2/orgs/1/traffic_flows/async_queries",
+                       json=json_data)
+
+    requests_mock.get(
+        "https://127.0.0.1:8443/api/v2/orgs/1/traffic_flows/async_queries/a89cb6ff-980f-46bf-b715-e021ce55c0db",
+        json=json_data_get)
+
+    requests_mock.get(
+        "https://127.0.0.1:8443/api/v2/orgs/1/traffic_flows/async_queries/d99b77b2-9ad4-4fe6-ac17-09ed634a47a4/download",
+        json=json_download_data)
+
+    response = illumio_traffic_analysis_command(mock_client, args)
+
+    assert response.outputs_prefix == "Illumio.TrafficFlows"
```

Comment:
This one function is testing too many things. Split them out so you have one core assertion per test.

Reply:
So for every assertion new test case?

Reply:
No need for a test for each of these assertions, the main thing is to identify a single unit under test and assert to confirm the expected output based on the given inputs.

To use this function as an example, I would split out at least two tests: one that confirms the expected raw output from calling the command function, and one that specifically checks the human readable output against the "golden" metadata output file.

Additionally, you might want to have a third test that highlights the difference between the `raw_response` and `outputs` fields in the command result by adding null values to the JSON response, or at least assert that in the command output test.

Reply:
Okay so I'll assert for the human-readable, raw_response & the one with null values as well

Reply:
I was thinking this would create duplication & maintenance problem, instead we can assert only for human readable and raw_response in the same test case.

Reply:
Duplication in tests is solved through parametrization or using common fixtures. For the human readable response, you shouldn't be calling the `_command` function anyway - just feed the expected result through the `prepare_` function and verify the expected output.

Reply:
Made the changes.